packages/context/test/unit/resolver.test.ts

```diff
@@ -320,3 +326,90 @@ function customAsyncDecorator(def: Object) {
     return f + ':' + b;
   });
 }
+
+describe('method injection', () => {
+  let ctx: Context;
+
+  before(function() {
+    ctx = new Context();
+    ctx.bind('foo').to('FOO');
+    ctx.bind('bar').to('BAR');
+  });
+
+  it('can resolve method arguments', () => {
+    class TestClass {
+      test(@inject('foo') foo: string) {
+        return `hello ${foo}`;
+      }
+    }
+
+    const t = invokeMethod(TestClass.prototype, 'test', ctx);
+    expect(t).to.eql('hello FOO');
+  });
+
+  it('can resolve method arguments with instance', () => {
```

Comment:
I find this confusing. Are you trying to verify that the method is invoked on the correct instance object? Perhaps the test should verify the contents of `this` instead of using a non-intuitive workaround of setting a property?

```ts
let invokedInstance = undefined;

class TestClass {
  test() {
    invokedInstance = this;
  }
}

const inst = new TestClass();
invokeMethod(inst, 'test', ctx);
expect(invokedInstance).to.equal(inst);
```

Reply:
(Same comment probably applies to "async" tests below too.)